### ISSUE TITLE
Failing preflight check for cors. Works outside of cypress-chrome for the same, non-cypress-ized browser (chrome) and a different browser (firefox).

### ISSUE THREAD

**Issue by robmoore-i** [cypress-io/cypress#2568]
### Current behavior:

- Preflight check (http OPTIONS request) fails with the following error shown in the console.

```
Failed to load http://localhost:8000/endpoint: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:3000' is therefore not allowed access. The response had HTTP status code 500.
```
 
Context:

- localhost:8000 is backend which serves json.
- localhost:3000 is the react frontend, using an XMLHttpRequest to fetch some data. That is the request that fails.

- The backend passes the following (python) integration test:

```js
def endpoint_options():
    headers = { "access-control-request-method": "GET" , "access-control-request-headers": "Content-Type" }
    options = requests.options("http://localhost:8000/lessonnames", headers=headers)
    assert_that(options.headers["Access-Control-Allow-Origin"]).is_equal_to("http://localhost:3000")
    assert_that(options.headers["Access-Control-Allow-Methods"]).is_equal_to("GET")
    assert_that(options.headers["Access-Control-Allow-Headers"]).is_equal_to("Content-Type")
    assert_that(options.status_code).is_equal_to(200)
```

- Using logs immediately before the endpoint returns a response, I've also found that the response being sent from the backend to cypress for the preflight check is this:

```
HTTP/1.x 200 OK
Access-Control-Allow-Origin: http://localhost:3000
Access-Control-Allow-Methods: GET
Access-Control-Allow-Headers: access-control-allow-origin
```

The cypress version of the chrome web browser however, shows this as the response (using the network tab of the chrome debugger).

```
HTTP/1.1 500 Internal Server Error
content-type: text/html; charset=utf-8
Vary: Accept-Encoding
Content-Encoding: gzip
Date: Sat, 06 Oct 2018 19:45:49 GMT
Connection: keep-alive
Transfer-Encoding: chunked
```

### Desired behavior:

The preflight cors check should instead receive the response that is sent by the backend, that is:

```
HTTP/1.x 200 OK
Access-Control-Allow-Origin: http://localhost:3000
Access-Control-Allow-Methods: GET
Access-Control-Allow-Headers: access-control-allow-origin
```

In doing this, the preflight check would succeed and the request be allowed.

### Steps to reproduce:

Failing cypress test is at https://github.com/robknows/Melange in the test/cyprus/integration directory.

The format is that there is a frontend directory with react stuff, a backend directory with a json serving rest api and a test directory for full, unmocked, end to end tests on locally running instances. That is what I'm hoping to try out cypress for, because there's also an interest to perhaps use it at work.

If you like, you can pull master from there and run the application and cypress yourself. That would be a hassle for you because you'd need to install a few things, like the q programming language (version 3.5), node, various react things, etc. Not to mention the fact I'm the only one working on it so I don't have CI or an automated build. No, I'd rather you didn't do loads of work because the solution is probably simple.

Instead, you could tell me how preflight CORS checking in the cypress-ized chrome browser is different from the ordinary chrome browser (chrome version 65). With that knowledge I could probably fix my problem or create a workaround. To disable CORS would make the end to end tests pointless really, so I'd rather not do that.

### Versions

Node 10.11.0

Cyprus 3.1.0

Google Chrome 65.0.3325.181

### Thanks for any help if you have time

**Comment by jennifer-shehane** [comment]
How are you running your Cypress tests exactly? Trying to verify what you mean by the Cypress-ized browser - is this after `cypress open` or `cypress run`?

**Comment by robknows** [comment]
By cypress-ized browser I mean the chrome browser that is opened by cyprus during the running of a test. Either by `cypress open` or `cypress run`.

Thanks for improving the issue name. Sorry for the slow reply.

**Comment by robknows** [comment]
If I use Chromium it actually works for me at the moment

**Comment by jennifer-shehane** [comment]
I see that you are using an older version of Cypress. Before I dive in to completely recreate the issue, could you update to the [current version of Cypress](https://on.cypress.io/changelog) and let me know if this is still happening for you? Your issue may have already been fixed. Thanks!

**Comment by robmoore-i** [comment]
Hey Jennifer, I can't remember if I ever switched off Chromium for this project, but that was never an issue and I never encountered this problem again. I'm going to close this issue and if it's really an issue I'm sure someone else will re-raise it.

Cheers,

Rob

**Comment by Santosh-Lodhi** [comment]
Hi I have CORS issue in chrome 

> Access to XMLHttpRequest at 'http://server:8080/api/update' from origin 'http://localhost:4200' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: The 'Access-Control-Allow-Origin' header contains the invalid value ''.

I am facing CORS issue with XMLHttpRequest I am using XHR because I need to pass form data on network, not able to do with cy.request(), is there any way to pass form data on network with cy.request() ?

**Comment by otalandim** [comment]
> I am facing CORS issue with XMLHttpRequest I am using XHR because I need to pass form data on network, not able to do with cy.request(), is there any way to pass form data on network with cy.request() ?

what's the problem?

**Comment by jennifer-shehane** [comment]
If you're experiencing a bug similar to this in Cypress, please [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a fully reproducible example that we can run. There may be a specific edge case with the issue that we need more detail to fix.
